apps/fz_http/lib/fz_http_web/controllers/auth_controller.ex

```diff
@@ -48,6 +49,29 @@ defmodule FzHttpWeb.AuthController do
     end
   end
 
+  def callback(conn, params) do
+    %{"provider" => provider} = params
+
+    with {:ok, tokens} <- OpenIDConnect.fetch_tokens(String.to_atom(provider), params),
+         {:ok, claims} <- OpenIDConnect.verify(String.to_atom(provider), tokens["id_token"]) do
```

Comment:
Probably should have a list of providers somewhere and always use `to_existing_atom`

Or even build a multi clause function with some macro to do the conversion directly.

Reply:
@princemaple The challenge is the list of providers will be configurable by the user at runtime. Ultimately it'll be defined in `/etc/firezone/firezone.rb` -- I think we're hoping to address #501 using a more general approach here so that we can support any OIDC-based provider.

`String.to_atom/1` is likely [problematic here though](http://www.moxleystratton.com/when-to-use-elixirs-string-to-atom/) because `provider` is end-user-supplied, right? Hmm. We may have to hack the OpenIDConnect module to support loading config from maps vs keyword lists?